ambry-store/src/main/java/com/github/ambry/store/StoreMetrics.java

```diff
@@ -231,14 +232,25 @@ public StoreMetrics(String prefix, MetricRegistry registry) {
         byteBufferForAppendTotalCountGauge);
   }
 
-  void initializeIndexGauges(String storeId, final PersistentIndex index, final long capacityInBytes) {
+  void initializeIndexGauges(String storeId, final PersistentIndex index, final long capacityInBytes,
+      BlobStoreStats blobStoreStats) {
     String prefix = storeId + SEPARATOR;
     Gauge<Long> currentCapacityUsed = index::getLogUsedCapacity;
     registry.register(MetricRegistry.name(Log.class, prefix + "CurrentCapacityUsed"), currentCapacityUsed);
     Gauge<Double> percentageUsedCapacity = () -> ((double) index.getLogUsedCapacity() / capacityInBytes) * 100;
     registry.register(MetricRegistry.name(Log.class, prefix + "PercentageUsedCapacity"), percentageUsedCapacity);
     Gauge<Long> currentSegmentCount = index::getLogSegmentCount;
     registry.register(MetricRegistry.name(Log.class, prefix + "CurrentSegmentCount"), currentSegmentCount);
+    Gauge<Long> currentInvalidDataSize = () -> {
+      long invalidDataSize = -1;
+      try {
+        invalidDataSize = index.getLogUsedCapacity() - blobStoreStats.getValidSize(
```

Comment:
what is the frequency of calling this method? `getValidSize` should be a small memory operation, but we never test it on real prod cluster before, maybe we want to also add a metric for getValidSize as well.

Reply:
with Invalid data, we can infer the valid data. Since we care about invalid data more, so invalid data is emitted.

Reply:
The frequency might be high. This is a valid concern. Let me think about this.

Reply:
it shouldn't take long to finish this method since it's all on memory. I think adding a metric in StoreMetrics and record it every time we call it would just do the work. If the latency is too high, we can think of something else.

Reply:
OK. Let's try. This is not in critical path, it should be OK.